### PR TITLE
Added grok support #1466

### PR DESCRIPTION
**Pull request by aldettinger**
[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

File: extensions/grok/deployment/src/main/java/org/apache/camel/quarkus/component/grok/deployment/GrokProcessor.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    private static final String FEATURE = "camel-grok";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    NativeImageResourceBuildItem registerDefaultPatternsNativeResources() {
+        return new NativeImageResourceBuildItem("patterns/patterns");
```
Comment (ppalaga) [inline extensions/grok/deployment/src/main/java/org/apache/camel/quarkus/component/grok/deployment/GrokProcessor.java:34]: I heart of grok for the first time today, but anyway, isn't `patterns/patterns` a typo?

Regardless whether it is correct, wouldn't it be worthwhile to document path?

Shouldn't the itest cover it somehow (I cannot see `patterns/patterns` folder in the itest)?

Comment (aldettinger) [reply extensions/grok/deployment/src/main/java/org/apache/camel/quarkus/component/grok/deployment/GrokProcessor.java:34]: Thanks for raising @ppalaga.

camel-grok comes with what the [camel doc](https://camel.apache.org/components/latest/dataformats/grok-dataformat.html#_predefined_patterns) calls `predefined patterns`, under the hood it's [java-grok default patterns](https://github.com/thekrakken/java-grok/blob/master/src/main/resources/patterns/patterns). Grok default patterns includes things like mac, ip, number and so on.

That said, `patterns/patterns` may be a bit vague but is not under our control. The test coverage must be correct as we are using mac, ip and so on in itests. Concerning the doc, I would let it in camel, maybe just clarify the predefined patterns section.

WDYT ?

Comment (ppalaga) [reply extensions/grok/deployment/src/main/java/org/apache/camel/quarkus/component/grok/deployment/GrokProcessor.java:34]: Oh, I see, so, `patterns/patterns` is not intended to be used by end users, right? All is fine then. Thanks for explaining.

Comment (aldettinger) [reply extensions/grok/deployment/src/main/java/org/apache/camel/quarkus/component/grok/deployment/GrokProcessor.java:34]: Not for end user indeed. Thanks for asking, this exchange will help me to improve the camel doc a bit and may help users in the future.

**Pull request merged**